File: fixtures/scale/kotlin/medium/src/calc/lexer/Symbols.kt
package calc.lexer

fun symbolTypeOf(c: Char): TokenType? = when (c) {
    '+' -> TokenType.PLUS
    '-' -> TokenType.MINUS
    '*' -> TokenType.STAR
    '/' -> TokenType.SLASH
    '^' -> TokenType.CARET
    '%' -> TokenType.PERCENT
    '(' -> TokenType.LPAREN
    ')' -> TokenType.RPAREN
    ',' -> TokenType.COMMA
    '=' -> TokenType.EQUALS
    else -> null
}

File: fixtures/scale/dart/medium/lib/lexer/symbols.dart
import 'token_type.dart';

TokenType? symbolTypeOf(String c) {
  switch (c) {
    case '+':
      return TokenType.plus;
    case '-':
      return TokenType.minus;
    case '*':
      return TokenType.star;
    case '/':
      return TokenType.slash;
    case '^':
      return TokenType.caret;
    case '%':
      return TokenType.percent;
    case '(':
      return TokenType.lparen;
    case ')':
      return TokenType.rparen;
    case ',':
      return TokenType.comma;
    case '=':
      return TokenType.equals;
    default:
      return null;
  }
}
